pilot/pkg/config/aggregate/config.go

```diff
@@ -107,28 +107,24 @@ func (cr *store) Get(typ config.GroupVersionKind, name, namespace string) *confi
 }
 
 // List all configs in the stores.
-func (cr *store) List(typ config.GroupVersionKind, namespace string) ([]config.Config, error) {
+func (cr *store) List(typ config.GroupVersionKind, namespace string) []config.Config {
 	if len(cr.stores[typ]) == 0 {
-		return nil, nil
+		return nil
```

Comment:
```suggestion
		return nil
```

If I understand  you right this should basically never happen, right? Can we throw a trace/debug log in all of these `List() > if then return nil` branches just so it would be easier to detect on the off chance something weird happened?